### ISSUE TITLE
Ecotax and Cost price decimals are truncated on BO when the product has combinations

### ISSUE THREAD

**Issue by itroom-agence** [PrestaShop/PrestaShop#14093]
**Describe the bug**
When we add an ecotax or a cost price with decimals to a product with combinations, it is truncated.

**To Reproduce**

1. Go to the BO -> Products
2. Edit a product with combinations
3. Combinations tab -> Edit a combination
4. Change the ecotax and the cost price for a price with decimals: 1.2345 
5. Save and leave the page
6. Edit the same product, change the name, save and leave the page
7. Edit the same product, see the combination edited before
8. The decimals for the ecotax and cost price disappeared

** Screerecord **

https://drive.google.com/file/d/1B14gj0N4BFKpuZ6iI_eA_YgiYWUDs9gp/view?usp=sharing

** Additional information **
PrestaShop version: 1.7.5 -> 1.7.6, 1.7.7

Add $ ecotax = str_replace (',', '.', $ Ecotax); line 1859 and 1952.

**Comment by mariem-abid** [comment]
Hi @itroom-agence,

I manage to reproduce the issue with PS:1.7.5.2 and PS 1.7.6.0beta1.
I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!

**Comment by sarahdib** [comment]
Hi @itroom-agence, 
I manage to reproduce this issue : 
https://drive.google.com/file/d/1A9z25EkRM5QkpW4pkXfp7eX9GzOmY5lA/view

Would you be willing to make a pull request on GitHub with your code suggestion?
https://github.com/PrestaShop/PrestaShop/tree/develop

Thank you

**Comment by marionf** [comment]
@sarahdib your video isn't available anymore

I tried to reproduce this issue on 1.7.7.x without success

@khouloudbelguith Could you try to reproduce it and link a video in the description if you reproduce it ?

Thanks

**Comment by khouloudbelguith** [comment]
Hi @sarahdib @marionf,

I did not manage to reproduce the issue with PS1770rc1build2.
I attached a screen record
https://drive.google.com/file/d/1xUV7SFX5CucN1bCca7yIYunuu2y-AHoI/view?usp=sharing

Thanks!

**Comment by marionf** [comment]
I reproduced the issue on 1.7.7.0, I edited the steps to reproduce and added a screenrecord

**Comment by PululuK** [comment]
Hi @itroom-agence 

I had the same issue, your proposal seems to work.
PR: https://github.com/PrestaShop/PrestaShop/pull/22631

Thank you !

**Comment by khouloudbelguith** [comment]
Hi @marionf

I think this issue is a duplicate of #9972
We have already a PR: https://github.com/PrestaShop/PrestaShop/pull/20144#issuecomment-662985378

What do you think?

Thank you!

**Comment by PululuK** [comment]
Hi @khouloudbelguith 

It's not the same issue and the two PRs are therefore different.
This is the ecotax issue.

Thanks

**Comment by mathieuhen** [comment]
Yes, for ecotax and for wholesale_price, there is the same issue. I can reproduce both. 

I think I know why you can't reproduce the bug. Personally my Prestashop is in French. So the decimal separator is a comma. 
If I switch my Prestashop profile to English, then I don't have the problem anymore since the decimal separator is the point.
So if you want reproduce the bug, switch your Prestashop interface in French

**Comment by alejandrocastanon** [comment]
The bug still on Prestashop 8.0.4

Any idea how to solve it?

**Comment by mathieuhen** [comment]
personally I solved it with an override of the Product class. 

I added this function

public function updateAttribute(
        $id_product_attribute,
        $wholesale_price,
        $price,
        $weight,
        $unit,
        $ecotax,
        $id_images,
        $reference,
        $ean13,
        $default,
        $location = null,
        $upc = null,
        $minimal_quantity = null,
        $available_date = null,
        $update_all_fields = true,
        array $id_shop_list = [],
        $isbn = '',
        $low_stock_threshold = null,
        $low_stock_alert = false,
        $mpn = null
    ) {
        
        $wholesale_price = str_replace(',', '.', $wholesale_price); 

        return parent::updateAttribute(
            $id_product_attribute,
            $wholesale_price,
            $price,
            $weight,
            $unit,
            $ecotax,
            $id_images,
            $reference,
            $ean13,
            $default,
            $location,
            $upc,
            $minimal_quantity,
            $available_date,
            $update_all_fields,
            $id_shop_list,
            $isbn,
            $low_stock_threshold,
            $low_stock_alert,
            $mpn
        );
        
    }

**Comment by alejandrocastanon** [comment]
@mathieuhen

Works like a charm. Thanks for your help

**Comment by ckdapps** [comment]
could some one give me correct process to fix this ? issue is clearly when the inteface separator in a comma instead of a point. i dont manage to overide on OS8.1.7. no impact, everytime a save the product, the ecotax get truncated

thanks all


>personally I solved it with an override of the Product class.
> 
> I added this function
> 
> public function updateAttribute( $id_product_attribute, $wholesale_price, $price, $weight, $unit, $ecotax, $id_images, $reference, $ean13, $default, $location = null, $upc = null, $minimal_quantity = null, $available_date = null, $update_all_fields = true, array $id_shop_list = [], $isbn = '', $low_stock_threshold = null, $low_stock_alert = false, $mpn = null ) {
> 
> ```
>     $wholesale_price = str_replace(',', '.', $wholesale_price); 
> 
>     return parent::updateAttribute(
>         $id_product_attribute,
>         $wholesale_price,
>         $price,
>         $weight,
>         $unit,
>         $ecotax,
>         $id_images,
>         $reference,
>         $ean13,
>         $default,
>         $location,
>         $upc,
>         $minimal_quantity,
>         $available_date,
>         $update_all_fields,
>         $id_shop_list,
>         $isbn,
>         $low_stock_threshold,
>         $low_stock_alert,
>         $mpn
>     );
>     
> }
> ```

**Comment by Hlavtox** [comment]
Closed, not relevant anymore.
